Revamp merging

ajeddeloh opened the pull request:
Ready for a first round of review.

jlebon commented on config/merge/merge_test.go:
```diff
@@ -0,0 +1,226 @@
+
+	tests := []test{
+		{
+			// case 1: appending empty configs is empty
```
Minor: you mean `merging` here right? :trollface:

jlebon commented on config/merge/merge_test.go:
```diff
@@ -0,0 +1,226 @@
+		out := MergeStruct(in1v, in2v).Interface().(types.Config)
+
+		assert.Equal(t, test.out, out, "#%d bas merge", i)
+	}
```
Would be good to have a negative test too to verify the "merging can fail" path. Though I'm not sure if you were planning to add this sort of test at a higher level.

ajeddeloh commented:
Fixed typos, add test for failing on bad merge, added code to fail on bad merge (whoops), added a minor quality of life improvement to the test runner I needed when adding said test.
